### PR TITLE
ckb-next: fix audio by adding libpulseaudio

### PR DESCRIPTION
**Pull request by virchau13**
This fixes audio by adding libpulseaudio.

###### Motivation for this change
Without this change, ckb-next doesn't support proper audio detection; e.g. the mute light does not work. This also adds a flag to turn off libpulseaudio inclusion in the package (enableLibpulseaudio).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kierdavis — APPROVED** [review]
Is there any reason why someone would want to disable pulseaudio support now?

**Comment by virchau13** [comment]
> Is there any reason why someone would want to disable pulseaudio support now?

None that I can think of. ¯\_(ツ)_/¯

**Comment by virchau13** [comment]
Sorry for the bother @SuperSandro2000, is there anything else I need to do?

File: pkgs/tools/misc/ckb-next/default.nix [hunk 1]
```diff
@@ -1,5 +1,6 @@
-{ lib, mkDerivation, fetchFromGitHub, substituteAll, udev
-, pkg-config, qtbase, cmake, zlib, kmod, libXdmcp, qttools, qtx11extras, libdbusmenu }:
+{ lib, mkDerivation, fetchFromGitHub, substituteAll, udev, stdenv
+, pkg-config, qtbase, cmake, zlib, kmod, libXdmcp, qttools, qtx11extras, libdbusmenu
+, enableLibpulseaudio ? stdenv.isLinux, libpulseaudio ? null }:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/ckb-next/default.nix:3]:
```suggestion
, enableLibpulseaudio ? stdenv.isLinux, libpulseaudio
}:
```

File: pkgs/tools/misc/ckb-next/default.nix [hunk 2]
```diff
@@ -1,5 +1,7 @@
-{ lib, mkDerivation, fetchFromGitHub, substituteAll, udev
-, pkg-config, qtbase, cmake, zlib, kmod, libXdmcp, qttools, qtx11extras, libdbusmenu }:
+{ lib, mkDerivation, fetchFromGitHub, substituteAll, udev, stdenv
+, pkg-config, qtbase, cmake, zlib, kmod, libXdmcp, qttools, qtx11extras, libdbusmenu
+, enableLibpulseaudio ? stdenv.isLinux, libpulseaudio
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/ckb-next/default.nix:3]:
```suggestion
, withPulseaudio ? stdenv.isLinux, libpulseaudio
```

File: pkgs/tools/misc/ckb-next/default.nix [hunk 3]
```diff
@@ -20,7 +22,7 @@ mkDerivation rec {
     qttools
     qtx11extras
     libdbusmenu
-  ];
+  ] ++ lib.optional enableLibpulseaudio libpulseaudio;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/ckb-next/default.nix:25]:
```suggestion
  ] ++ lib.optional withPulseaudio libpulseaudio;
```

**Pull request merged**